lib/Twig/Environment.php

```diff
@@ -326,15 +338,13 @@ public function loadTemplate($name, $index = null)
         }
 
         if (!class_exists($cls, false)) {
-            if (false === $cache = $this->getCacheFilename($name)) {
-                eval('?>'.$this->compileSource($this->getLoader()->getSource($name), $name));
-            } else {
-                if (!is_file($cache) || ($this->isAutoReload() && !$this->isTemplateFresh($name, filemtime($cache)))) {
-                    $this->writeCacheFile($cache, $this->compileSource($this->getLoader()->getSource($name), $name));
-                }
+            $key = $this->cache->getCacheKey($cls, $this->templateClassPrefix);
 
-                require_once $cache;
+            if (!$this->cache->has($key) || ($this->isAutoReload() && !$this->isTemplateFresh($name, $this->cache->getTimestamp($key)))) {
```

Comment:
how about moving the auto reload property into the cache, which lets it decide if it should autoreload or not in the `has` method ? And also moving the `isTemplateFresh` into the cache too ?